spec/factories/grant.rb

```diff
@@ -61,9 +65,19 @@
       end
     end
 
-    factory :open_grant,                               traits: %i[open with_questions with_users]
-    factory :closed_grant,                             traits: %i[closed with_questions with_users]
+    trait :bypass_validations do
+      to_create { |grant| grant.save(validate: false) }
+    end
 
+    factory :draft_grant,                              traits: %i[draft]
+    factory :published_grant,                          traits: %i[published]
+    factory :open_grant_with_users_and_questions,      traits: %i[open with_questions with_users]
+    factory :closed_grant_with_users_and_questions,    traits: %i[closed with_questions with_users]
+    factory :published_open_grant,                     traits: %i[published open]
+    factory :published_closed_grant,                   traits: %i[published closed]
+    factory :completed_closed_grant,                   traits: %i[completed closed]
```

Comment:
could be called completed_grant since completed grants are closed by definition.